packages/contracts-bedrock/src/L2/XForkContractsManager.sol

```diff
@@ -4,13 +4,121 @@ pragma solidity 0.8.15;
+        address l1BlockAttributesImplementation; // 9: L1Block
+        address l2ToL1MessagePasserImplementation; // 10: L2ToL1MessagePasser
+        address optimismMintableERC721FactoryImplementation; // 11: OptimismMintableERC721Factory
+        address baseFeeVaultImplementation; // 12: BaseFeeVault
+        address l1FeeVaultImplementation; // 13: L1FeeVault
+        address operatorFeeVaultImplementation; // 14: OperatorFeeVault
+        address schemaRegistryImplementation; // 15: SchemaRegistry
+        address easImplementation; // 16: EAS
+    }
+
+    address internal immutable LEGACY_MESSAGE_PASSER_IMPLEMENTATION;
+    address internal immutable DEPLOYER_WHITELIST_IMPLEMENTATION;
+    address internal immutable L2_CROSS_DOMAIN_MESSENGER_IMPLEMENTATION;
+    address internal immutable GAS_PRICE_ORACLE_IMPLEMENTATION;
+    address internal immutable L2_STANDARD_BRIDGE_IMPLEMENTATION;
+    address internal immutable SEQUENCER_FEE_WALLET_IMPLEMENTATION;
+    address internal immutable OPTIMISM_MINTABLE_ERC20_FACTORY_IMPLEMENTATION;
+    address internal immutable L1_BLOCK_NUMBER_IMPLEMENTATION;
+    address internal immutable L2_ERC721_BRIDGE_IMPLEMENTATION;
+    address internal immutable L1_BLOCK_ATTRIBUTES_IMPLEMENTATION;
+    address internal immutable L2_TO_L1_MESSAGE_PASSER_IMPLEMENTATION;
+    address internal immutable OPTIMISM_MINTABLE_ERC721_FACTORY_IMPLEMENTATION;
+    address internal immutable BASE_FEE_VAULT_IMPLEMENTATION;
+    address internal immutable L1_FEE_VAULT_IMPLEMENTATION;
+    address internal immutable OPERATOR_FEE_VAULT_IMPLEMENTATION;
+    address internal immutable SCHEMA_REGISTRY_IMPLEMENTATION;
+    address internal immutable EAS_IMPLEMENTATION;
+
+    constructor(Input memory _input) {
+        if (
+            _input.legacyMessagePasserImplementation == address(0)
+                || _input.deployerWhitelistImplementation == address(0)
+                || _input.l2CrossDomainMessengerImplementation == address(0)
+                || _input.gasPriceOracleImplementation == address(0) || _input.l2StandardBridgeImplementation == address(0)
+                || _input.sequencerFeeWalletImplementation == address(0)
+                || _input.optimismMintableERC20FactoryImplementation == address(0)
+                || _input.l1BlockNumberImplementation == address(0) || _input.l2ERC721BridgeImplementation == address(0)
+                || _input.l1BlockAttributesImplementation == address(0)
+                || _input.l2ToL1MessagePasserImplementation == address(0)
+                || _input.optimismMintableERC721FactoryImplementation == address(0)
+                || _input.baseFeeVaultImplementation == address(0) || _input.l1FeeVaultImplementation == address(0)
+                || _input.operatorFeeVaultImplementation == address(0) || _input.schemaRegistryImplementation == address(0)
+                || _input.easImplementation == address(0)
+        ) {
+            revert InvalidInput();
+        }
```

Comment:
Do we need this? maybe we can assume these addresses will be valid, I dont see a check like this on the OPCM, we can define this as an assumption on the spec